lightning/src/routing/router.rs

```diff
@@ -71,6 +71,26 @@ pub struct Route {
 	pub paths: Vec<Vec<RouteHop>>,
 }
 
+impl Route {
+	/// Returns the total amount of fees paid on this Route.
+	/// This doesn't include any extra payment made to the recipient,
+	/// which can happen in excess of the amount passed to `get_route`'s `final_value_msat`.
+	pub fn get_total_fees(&self) -> u64 {
+		// Do not count last hop of each path since that's the full value of the payment
+		return self.paths.iter()
+			.flat_map(|path| path.split_last().unwrap().1)
+			.map(|hop| &hop.fee_msat)
+			.sum();
+	}
+	/// Returns the total amount paid on this Route, excluding the fees.
+	pub fn get_total_amount(&self) -> u64 {
```

Comment:
Idk if it's worth noting for these methods that they'll panic for 0-length routes?